packages/pluggableWidgets/datagrid-web/src/components/Table.tsx

```diff
@@ -82,10 +83,18 @@ export function Table<T>(props: TableProps<T>): ReactElement {
                 disableResizing: !column.resizable,
                 disableFilters: column.filterable === "no",
                 sortType: (rowA: Row<{ item: T }>, rowB: Row<{ item: T }>, columnId: IdType<object>): number => {
-                    const valueA = props.valueForFilterSort(rowA.values[columnId], Number(columnId)) || "";
-                    const valueB = props.valueForFilterSort(rowB.values[columnId], Number(columnId)) || "";
+                    const valueA = props.valueForSort(rowA.values[columnId], Number(columnId)) || "";
+                    const valueB = props.valueForSort(rowB.values[columnId], Number(columnId)) || "";
                     // Values should always be sorted in ASC mode https://github.com/tannerlinsley/react-table/pull/2504
-                    return valueA.localeCompare(valueB);
+                    if (typeof valueA === "string" && typeof valueB === "string") {
+                        return valueA.localeCompare(valueB);
+                    } else if (typeof valueA === "boolean" && typeof valueB === "boolean") {
+                        // True first
+                        return valueA === valueB ? 0 : valueA ? -1 : 1;
+                    } else if (valueA instanceof Date && valueB instanceof Date) {
+                        return (valueA as Date).getTime() - (valueB as Date).getTime();
```

Comment:
Do we still has to cast `as Date` since we said in line 94 it is an instaceof Date ?

Reply:
If you want to be able to get the time yes..